modules/idaithalam/src/main/java/io/virtualan/idaithalam/contract/IdaithalamExecutor.java

```diff
@@ -109,16 +101,23 @@ public static int validateContract(String featureHeading)
      * @throws UnableToProcessException the unable to process exception
      */
     public static int validateContract(String featureHeading, ApiExecutorParam apiExecutorParam)
-        throws UnableToProcessException {
+            throws UnableToProcessException {
         byte exitStatus;
         try {
-            String fileIndex = UUID.randomUUID().toString();
             VirtualanClassLoader classLoaderParnet = new VirtualanClassLoader(IdaithalamExecutor.class.getClassLoader());
             ExecutionClassloader classLoader = addConfToClasspath(classLoaderParnet, apiExecutorParam.getOutputDir());
-            generateFeatureFile(classLoader, apiExecutorParam);
+            /** @author Oliver Glas. */
+            if (apiExecutorParam.getExecution() == null || apiExecutorParam.getExecution().equalsIgnoreCase("execute") == false) {
+                generateFeatureFile(classLoader, apiExecutorParam);
+            }
+            if (apiExecutorParam.getExecution() != null && apiExecutorParam.getExecution().equalsIgnoreCase("generate")) {
```

Comment:
move the condition inside the generate after  line 111

Reply:
Done.